### PR TITLE
crud of province and sector

### PR DESCRIPTION
**Pull request by PhilippeTsongo**
crud of province and sector

### CODE REVIEW

**Comment by PhilippeTsongo** [comment]
branch where the features(province and sector) are running: _Philippe_

### PR registers provinces and sector ### 
**How to test**
_register a  province_
method: post
This PR is tested with this endpoint http://127.0.0.1:8000/api/province 
required field: **name**
 
_register a sector_
method: post
This PR is tested with this endpoint http://127.0.0.1:8000/api/sector 
required field: **name, distritcId**

_List of province_
method: get
This PR is tested with this endpoint http://127.0.0.1:8000/api/province


_List of sector_
method: get
This PR is tested with this endpoint http://127.0.0.1:8000/api/sector


_updater a  province_
method: put
This PR is tested with this endpoint http://127.0.0.1:8000/api/province/1

(parameter here in this request is the is the id of the province to be updated)
required field: **name**
  
_updater a  sector_
method: put
This PR is tested with this endpoint http://127.0.0.1:8000/api/province/1

(parameter here in this request is the is the id of the sector to be updated)
required field: **name, districtId**


_delete a  province_
method: delete
This PR is tested with this endpoint http://127.0.0.1:8000/api/province/1

(parameter here in this request is the is id of the province to be deleted)

_delete a  sector_
method: delete
This PR is tested with this endpoint http://127.0.0.1:8000/api/sector/1

(parameter here in this request 1 is the ID of the sector to be deleted)

File: app/Http/Controllers/ProvinceController.php [hunk 1]
```diff
@@ -8,19 +8,22 @@
...
+     * */
     public function index()
     {
-        //
+        $provinces = Province::orderBy('id', 'DESC')
+                    ->get();
+
+        return response()->json([
+            'provinces' => $provinces,
+        ]);
```
Comment (iyaremye-faustin) [inline app/Http/Controllers/ProvinceController.php:21]: why are you not sending the status code to help frontend know what the status response returned? Please check main and reuse apiResponse already implemented for easing this.

File: app/Http/Controllers/ProvinceController.php [hunk 2]
```diff
@@ -30,56 +33,81 @@ public function create()
...
     }
 
     /**
      * Show the form for editing the specified resource.
      *
-     * @param  \App\Models\Province  $province
-     * @return \Illuminate\Http\Response
      */
     public function edit(Province $province)
```
Comment (iyaremye-faustin) [inline app/Http/Controllers/ProvinceController.php:72]: Remove the functions that you are not using

File: app/Http/Controllers/ProvinceController.php [hunk 3]
```diff
@@ -30,56 +33,81 @@ public function create()
...
      * Remove the specified resource from storage.
      *
-     * @param  \App\Models\Province  $province
-     * @return \Illuminate\Http\Response
      */
     public function destroy(Province $province)
     {
-        //
+        $province->delete();
+
```
Comment (iyaremye-faustin) [inline app/Http/Controllers/ProvinceController.php:107]: I think i will not be a good practice to delete a province but to edit it, this is not needed

File: app/Http/Controllers/SectorController.php [hunk 4]
```diff
@@ -8,19 +8,22 @@
...
-     */
+     * */
     public function index()
     {
-        //
+        $sector = Sector::orderBy('id', 'DESC')
+                    ->get();
+
+        return response()->json([
+            'sector' => $sector,
```
Comment (iyaremye-faustin) [inline app/Http/Controllers/SectorController.php:20]: Status code is needed here

File: app/Http/Controllers/SectorController.php [hunk 5]
```diff
@@ -30,56 +33,85 @@ public function create()
...
     }
 
     /**
      * Show the form for editing the specified resource.
      *
-     * @param  \App\Models\Sector  $sector
-     * @return \Illuminate\Http\Response
      */
     public function edit(Sector $sector)
     {
```
Comment (iyaremye-faustin) [inline app/Http/Controllers/SectorController.php:75]: Remove this function that you are not using

**Review by iyaremye-faustin — COMMENTED** [review]
Great work, work on some comments and swagger documentation and do rebase please

**Pull request merged**